Assets/CesiumForUnitySamples/Scripts/CesiumSamplesRequiresMagicLeap.cs

```diff
@@ -0,0 +1,243 @@
+            },
+            () =>
+            {
+                OpenXRSettings oxrSettings = OpenXRSettings.GetSettingsForBuildTargetGroup(BuildTargetGroup.Android);
+                MagicLeapRenderingExtensionsFeature renderFeature = oxrSettings.GetFeature<MagicLeapRenderingExtensionsFeature>();
+                renderFeature.enabled = true;
+                renderFeature.GlobalDimmerEnabled = true;
+            }),
+        // Make sure the Magic Leap Controller is enabled as an input source
+        new Case(
+            () =>
+            {
+                OpenXRSettings oxrSettings = OpenXRSettings.GetSettingsForBuildTargetGroup(BuildTargetGroup.Android);
+                OpenXRFeature[] interactionFeatures = oxrSettings.GetFeatures<OpenXRInteractionFeature>();
+                return interactionFeatures.Any(f => f.GetType() == typeof(MagicLeapControllerProfile) && f.enabled);
+            },
+            () =>
+            {
+                OpenXRSettings oxrSettings = OpenXRSettings.GetSettingsForBuildTargetGroup(BuildTargetGroup.Android);
+
+                OpenXRFeature[] interactionFeatures = oxrSettings.GetFeatures<OpenXRInteractionFeature>();
+                MagicLeapControllerProfile magicLeapProfile = interactionFeatures.FirstOrDefault(f => f.GetType() == typeof(MagicLeapControllerProfile)) as MagicLeapControllerProfile;
+                if (magicLeapProfile != null)
+                {
+                    magicLeapProfile.enabled = true;
+                }
+            }),
+        // We need to disable ML's handling of the far clip plane, so we can see distant terrain
+        // This property is private so we use SerializedProperty to access it
+        new Case(
+            () =>
+            {
+                OpenXRSettings oxrSettings = OpenXRSettings.GetSettingsForBuildTargetGroup(BuildTargetGroup.Android);
+                MagicLeapFeature mlFeature = oxrSettings.GetFeature<MagicLeapFeature>();
+                return mlFeature != null && mlFeature.enabled && mlFeature.FarClipPolicy == MagicLeapFeature.FarClipMode.None;
+            },
+            () =>
+            {
+                OpenXRSettings oxrSettings = OpenXRSettings.GetSettingsForBuildTargetGroup(BuildTargetGroup.Android);
+                MagicLeapFeature mlFeature = oxrSettings.GetFeature<MagicLeapFeature>();
+                mlFeature.enabled = true;
+
+                SerializedObject mlFeatureObject = new SerializedObject(mlFeature);
+                SerializedProperty farClipProperty = mlFeatureObject.FindProperty("farClipPolicy");
+                farClipProperty.intValue = (int)MagicLeapFeature.FarClipMode.None;
+                mlFeatureObject.ApplyModifiedPropertiesWithoutUndo();
+            })
+#endif // CESIUM_MAGIC_LEAP
+    };
+
+    static CesiumSamplesRequiresMagicLeap()
+    {
+        EditorApplication.playModeStateChanged += OnStateChanged;
+    }
+#endif // UNITY_2022_2_OR_NEWER
```

Comment:
Make sure to update for `UNITY_2022_3_11_OR_NEWER` (or whatever the correct define is?)